src/fields/u32/fp/wrapper.rs

```diff
@@ -58,6 +57,85 @@ impl Fp {
         fiat::fp_square(&mut result, &self.0);
         Self(result)
     }
+
+    fn inverse(&self) -> Option<Fp> {
+        if self == &Fp::zero() {
+            return None;
+        }
+
+        const LEN_PRIME: usize = 377;
+        const ITERATIONS: usize = (49 * LEN_PRIME + 57) / 17;
+
+        let mut a = fiat::FpNonMontgomeryDomainFieldElement([0; 12]);
+        fiat::fp_from_montgomery(&mut a, &self.0);
+        let mut d = 1;
+        let mut f: [u32; 13] = [0u32; 13];
+        fiat::fp_msat(&mut f);
+        let mut g = [0u32; 13];
+        let mut v = [0u32; 12];
+        let mut r: [u32; 12] = Fp::one().0 .0;
+        let mut i = 0;
+        let mut j = 0;
+
+        while j < 6 {
```

Comment:
I think this should be 12 instead of 6 (to review I was comparing with the `impl_bernstein_yang_invert` macro where j [goes up to `n_limbs`](https://github.com/RustCrypto/elliptic-curves/blob/master/primeorder/src/field.rs#L533))

Reply:
This is a good catch! Fixed by changing limbs to a constant.